File: 백준/Silver/17952. 과제는 끝나지 않아！/과제는 끝나지 않아！.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

var in = bufio.NewScanner(os.Stdin)

func nextInt() int {
	in.Scan()
	r := 0
	for _, c := range in.Bytes() {
		r *= 10
		r += int(c - '0')
	}
	return r
}

type Work struct {
	score, time int
}

func main() {
	in.Split(bufio.ScanWords)
	N := nextInt()
	ans := 0
	stack := []Work{}
	for i := 0; i < N; i++ {
		if nextInt() == 1 {
			score := nextInt()
			time := nextInt()
			if time == 1 {
				ans += score
			} else {
				stack = append(stack, Work{score, time - 1})
			}
		} else {
			if len(stack) == 0 {
				continue
			}
			w := stack[len(stack)-1]
			stack = stack[:len(stack)-1]
			if w.time == 1 {
				ans += w.score
			} else {
				stack = append(stack, Work{w.score, w.time - 1})
			}
		}
	}
	fmt.Println(ans)
}

File: 백준/Silver/17952. 과제는 끝나지 않아！/과제는 끝나지 않아！.java
import java.io.*;
import java.util.*;
public class Main{
    static class Work {
        int score, time;
        public Work(int score, int time) {
            this.score = score;
            this.time = time;
        }
    }

    public static void main(String[] args)throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
//        StringBuilder ans = new StringBuilder();
        int N = Integer.parseInt(br.readLine()),ans = 0;
        Stack<Work> stack = new Stack<>();
        for (int i = 0; i < N; i++){
            StringTokenizer st = new StringTokenizer(br.readLine());
            if (st.nextToken().equals("1")){
                int score = Integer.parseInt(st.nextToken())
                        , time = Integer.parseInt(st.nextToken());
                if (time == 1) ans += score;
                else stack.add(new Work(score, time-1));
            }else{
                if (stack.isEmpty()) continue;
                Work w = stack.pop();
                if (w.time == 1) ans += w.score;
                else stack.add(new Work(w.score, w.time-1));
            }
        }

        System.out.println(ans);
    }
}